File: data/multilang_comparison/program_cpp_vector/src/func1.cpp
#include "program_cpp_vector.hpp" 
Array* func1(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array225;
   if (pCounter > 0) {
      array225 = vars->data.at(--pCounter);
      array225->refC++;
      DEBUG_COPY(array225->id);
   } else {
      array225 = new Array();
      array225->size = 137;
      array225->refC = 1;
      array225->id = 225;
      array225->data = std::vector<unsigned int>(array225->size, 0);
      DEBUG_NEW(array225->id);
   }
   Array_param params0;
   params0.size = 1;
   std::vector<Array*> data_params0(1);
   params0.data = data_params0;
   params0.data.at(0) = array225;
   Array* array226 = func3(&params0, loopsFactor);
   DEBUG_RETURN(array226->id);
   if(params0.data.size() > 0) {
   	params0.data.clear();
   }
   if(PATH0 & 1) {
      Array* array229;
      if (pCounter > 0) {
         array229 = vars->data.at(--pCounter);
         array229->refC++;
         DEBUG_COPY(array229->id);
      } else {
         array229 = new Array();
         array229->size = 963;
         array229->refC = 1;
         array229->id = 229;
         array229->data = std::vector<unsigned int>(array229->size, 0);
         DEBUG_NEW(array229->id);
      }
      unsigned int loop66 = 0;
      unsigned int loopLimit66 = (50)/2 + 1;
      for(; loop66 < loopLimit66; loop66++) {
         for (int i = 0; i < array229->size; i++) {
             array229->data.at(i)--;
         }
         for (int i = 0; i < array226->size; i++) {
             if (array226->data.at(i) == 58) {
                 return array226;
             }
         }
      }
      Array_param params1;
      params1.size = 3;
      std::vector<Array*> data_params1(3);
      params1.data = data_params1;
      params1.data.at(0) = array225;
      params1.data.at(1) = array226;
      params1.data.at(2) = array229;
      Array* array230 = func10(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array230->id);
      if(params1.data.size() > 0) {
      	params1.data.clear();
      }
      array230->refC--;
      if(array230->refC == 0) {
         array230->data.clear();
         delete array230;
         DEBUG_FREE(array230->id);
      }
      array229->refC--;
      if(array229->refC == 0) {
         array229->data.clear();
         delete array229;
         DEBUG_FREE(array229->id);
      }
   }
   else {
      Array_param params1;
      params1.size = 2;
      std::vector<Array*> data_params1(2);
      params1.data = data_params1;
      params1.data.at(0) = array225;
      params1.data.at(1) = array226;
      Array* array231 = func4(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array231->id);
      if(params1.data.size() > 0) {
      	params1.data.clear();
      }
      array231->refC--;
      if(array231->refC == 0) {
         array231->data.clear();
         delete array231;
         DEBUG_FREE(array231->id);
      }
   }
   array225->refC--;
   if(array225->refC == 0) {
      array225->data.clear();
      delete array225;
      DEBUG_FREE(array225->id);
   }
   return array226;
}

File: data/multilang_comparison/program_go/program_go/src/func1.go
package main
import "math/rand"
var _ = rand.Intn
func Func1(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array225 *Array
   if pCounter > 0 {
       array225 = vars.Data[pCounter-1]
       pCounter--
       array225.RefC++
   } else {
       array225 = &Array{
           Size: 137,
           RefC: 1,
           Id: 225,
           Data: make([]uint32, 137),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array225
   var array226 *Array = Func3(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   if (PATH0 & 1) != 0 {
      var array229 *Array
      if pCounter > 0 {
          array229 = vars.Data[pCounter-1]
          pCounter--
          array229.RefC++
      } else {
          array229 = &Array{
              Size: 963,
              RefC: 1,
              Id: 229,
              Data: make([]uint32, 963),
          }
      }
      var loop66 int = 0;
      var loopLimit66 int = (50)/2 + 1;
      for ; loop66 < loopLimit66; loop66++ {
         for i := 0; i < array229.Size; i++ {
             array229.Data[i]--
         }
         for i := 0; i < array226.Size; i++ {
             if array226.Data[i] == 58 {
                 return array226
             }
         }
      }
      var params1 ArrayParam
      params1.Size = 3
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array225
      params1.Data[1] = array226
      params1.Data[2] = array229
      var array230 *Array = Func10(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array230.RefC--
      if array230.RefC == 0 {
          array230.Data = nil
          array230 = nil
      }
      array229.RefC--
      if array229.RefC == 0 {
          array229.Data = nil
          array229 = nil
      }
   
   } else {
      var params1 ArrayParam
      params1.Size = 2
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array225
      params1.Data[1] = array226
      var array231 *Array = Func4(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array231.RefC--
      if array231.RefC == 0 {
          array231.Data = nil
          array231 = nil
      }
   }
   array225.RefC--
   if array225.RefC == 0 {
       array225.Data = nil
       array225 = nil
   }
   return array226;
}

